check for a time.ParseError, add unit tests

bacongobbler opened the pull request:
refs #10

Commit:
fix(time): check explicitly for a time.ParseError

Commit:
tests(time): add unit tests

Commit:
style(time): go fmt

arschles commented on time/time.go:
```diff
@@ -3,7 +3,8 @@ package time
 
 // DeisDatetimeFormat is the standard date/time representation used in Deis.
-const DeisDatetimeFormat  string = "2006-01-02T15:04:05MST"
+const DeisDatetimeFormat string = "2006-01-02T15:04:05MST"
```
don't need the `string` after the const name here. same with the below

bacongobbler replied:
good points. Old code but isn't necessary yeah.

arschles commented on time/time.go:
```diff
@@ -27,7 +28,7 @@ func (t *Time) MarshalJSON() ([]byte, error) {
 // The time is expected to be in Deis' datetime format.
 func (t *Time) UnmarshalText(data []byte) (err error) {
```
can you remove the named return param here? it's shadowed below and hence not used

arschles commented on time/time.go:
```diff
@@ -27,7 +28,7 @@ func (t *Time) MarshalJSON() ([]byte, error) {
-	if err != nil {
+	if _, ok := err.(*time.ParseError); ok {
 		tt, err = time.Parse(PyOpenSSLTimeDateTimeFormat, string(data))
 	}
```
what about other errors? I'd like to be defensive here in case other errors ever are returned

arschles replied:
same with the below

bacongobbler replied:
I've refactored the function to explicitly return the error, which was the original intention. Does that look better to you?

arschles replied:
yup! and thanks for removing the named return param too

Commit:
ref(time): remove unnecessary string declarations

Commit:
fix(tests): remove named error param

This value was being shadowed and hence not used. Explicitly returning
the shadowed return value ensures this code performs as expected.

arschles commented on time/time_test.go:
```diff
@@ -0,0 +1,24 @@
+		"2006-01-02T15:04:05",
+	}
+	for _, goodTime := range goodTimeFormats {
+		if dummyTime.UnmarshalText([]byte(goodTime)) != nil {
```
for some extra sanity, can you split the formats and actual values apart, so the tests ensures that multiple times of the given format work?

bacongobbler replied:
certainly

arschles commented:
@bacongobbler left one final comment regarding tests. LGTM after that's addressed

Commit:
feat(time): embed time struct

This way functions like .Year() are available to the user without
having to run `t.Time.Year()`. now all they need to do is call
`t.Year()`.

bacongobbler commented:
Okay I split them out by time format. I also embedded the time struct so you no longer have to call `t.Time.Year()` every time you want to check the year of the object. Things like `time.Now()` still won't be available, but it should make the deis-specific time struct much more usable. I added some tests checking the Year to demonstrate its usage. It's also completely backwards compatible because you can still call `t.Time` to get the raw Time struct.

arschles commented:
:+1:

bacongobbler commented:
I will admit that this has gone out of scope of the original intention, so please let me know if you'd be happier with this all split out into different PRs.

arschles commented:
@bacongobbler I'm good with it as-is because you've added tests :)

bacongobbler commented:
tests make everyone happy <3
